### ISSUE TITLE
GSConnect fails to connect my Pixel after the last OS update

### ISSUE THREAD

**Issue by manti-by** [GSConnect/gnome-shell-extension-gsconnect#1209]
```
GSConnect: 43 (user)
GJS: 16405
Session: x11
OS: Ubuntu 20.04.3 LTS
GNOME Shell: 3.36.9
Meson: 0.53.2
Ninja: 1.10.0

/(?:cn|CN) ?= ?([^,\n]*)/.exec(...) is null

get@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/__init__.js:393:64
_initCertificate@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/backends/lan.js:182:13
start@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/backends/lan.js:440:18
_loadBackends@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/manager.js:251:25
start@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/manager.js:454:14
vfunc_startup@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:258:22
vfunc_handle_local_options@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:666:18
@/home/manti/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:729:17
```

**Comment by andyholmes** [comment]
It seems like there's probably a new version of `openssl` making the rounds with different output we're failing to parse. Someone with this new version may have to figure out the difference and add an extra if-then clause to account for this.

**Comment by manti-by** [comment]
> It seems like there's probably a new version of `openssl` making the rounds with different output we're failing to parse. Someone with this new version may have to figure out the difference and add an extra if-then clause to account for this.

Yep, seems like this is because I've updated OpenSSL from 1.0.2 to 1.1.1g. I'll maybe try to check this issue, but I don't know how)

**Comment by andyholmes** [comment]
These are the two commands we parse output from:

1. Generate fingerprint (sha256sum of DER data):

    https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/396ae953f965c5bbbc2a6e354c78b858b623f98c/src/service/__init__.js#L366

2. Extract device ID (CN):

    https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/396ae953f965c5bbbc2a6e354c78b858b623f98c/src/service/__init__.js#L387

The certificate and private/public key pair are generated using this command (although you can just run this on your existing certificate):

https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/396ae953f965c5bbbc2a6e354c78b858b623f98c/src/service/__init__.js#L337-L343

So if you can figure out how the output changes, it would just be a matter of something like:

```js
let match = /(?:cn|CN) ?= ?([^,\n]*)/.exec(stdout);

if (match[1] === null)
    match = /new-regexp/.exec(stdout);

this.__common_name = match[1];
```

It occurs to me that it may be a change in CLI arguments though, but either way this should help point you in the right direction. If you're not familiar with JavaScript or just need some help, do ask :slightly_smiling_face:

**Comment by andyholmes** [comment]
See also: https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1395#issuecomment-1159515032

**Comment by sfarbota** [comment]
I'm also getting this error. And it does not seem to be related to openssl in my case as openssl seems to be working perfectly fine otherwise.

This was working fine for me until I rebooted after making several changes on my laptop (applying a minor Ubuntu update, removing what I thought was a duplicate copy of glib2, probably a few other things).

And I should also note that in my case, this was not the only thing that broke. Pretty much all my Ubuntu user settings (desktop background, favorite apps, keyboard shortcuts, etc.) were wiped. And Google Chrome would start correctly either. But I was able to resolve all those other problems by reapplying/reinstalling them.

Any other ideas?

**Comment by realysy** [comment]
I got same error:
```log
get@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/__init__.js:392:35
_initCertificate@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/backends/lan.js:182:13
start@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/backends/lan.js:440:18
_loadBackends@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/manager.js:251:25
start@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/manager.js:454:14
vfunc_startup@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:258:22
vfunc_handle_local_options@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:666:18
@~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:729:17
```

And finally fix it by downgrading openssl from v3.0+ to 1.1.1f. The `openssl` command is available both before and after the downgrade.

---

Env:
* system: ubuntu 20.04
* shell version: 3.36.9, check by `gnome-shell --version`
* GSConnection extension version: 53, check by `cat ~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/metadata.json | grep '"version"'`

**Comment by ferdnyc** [comment]
@realysy 

Well, I certainly can't recommend running with OpenSSL 1.1.1f instead of 3.x _in general_. 

But since...
1. GSConnect 53 is very old and unsupported
2. Ubuntu 20.04 will only be supported (by Canonical, not us, we already don't support it) until May, and
3. [it looks like](https://bugs.launchpad.net/ubuntu/+source/openssl/1.1.1f-1ubuntu2.23) Ubuntu have been keeping their 1.1.1f builds patched against security vulnerabilities as they've been discovered

I guess if it works for you, then problem solved! 👍
